### PR TITLE
feat: Support transform animations for positioned components

### LINKED ISSUE #24133: [Bug]: V9 Menu/Popover on open missing motion transition
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
https://master--628d031b55e942004ac95df1.chromatic.com/?path=/docs/components-menu-menu--default
```

### Are you reporting Accessibility issue?

_No response_

### Reproduction

V9

## Actual Behavior
Menu open is just a binary on and off, would like a motion transition similar to the v7 contextual menu

## Expected Behavior
can the on open motion be similar to the Fluent v7 implementation? 
https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by ling1726**
Uses the [animation-composition](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-composition#browser_compatibility) to support sliding animation for `Popover` and `Menu` components.

Adds graceful degradation for browsers that do not support the animation-composition CSS property and I tested this with [Firefox 79](https://ftp.mozilla.org/pub/firefox/releases/).

Also adds reduced motion styles that don't use the transform - only opacity is changed

## Previous Behavior

Popover/Menu don't animate on enter

## New Behavior
proposed animation

Fixes #24133

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                      Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `87.735 kB`<br />`28.243 kB` |  `88.088 kB`<br />`28.341 kB` |     `353 B` <br />`98 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `86.074 kB`<br />`27.848 kB` |  `86.427 kB`<br />`27.972 kB` |    `353 B` <br />`124 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `206.425 kB`<br />`57.914 kB` | `207.813 kB`<br />`58.232 kB` | `1.388 kB` <br />`318 B`  |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr>                                                          |  `222.56 kB`<br />`59.204 kB` | `222.913 kB`<br />`59.349 kB` |    `353 B` <br />`145 B`  |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoButton.fixture.js'>InfoButton</abbr>                                                                        | `130.121 kB`<br />`39.785 kB` | `131.509 kB`<br />`40.129 kB` | `1.388 kB` <br />`344 B`  |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoLabel.fixture.js'>InfoLabel</abbr>                                                                          | `133.586 kB`<br />`40.852 kB` | `134.974 kB`<br />`41.185 kB` | `1.388 kB` <br />`333 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `130.848 kB`<br />`39.946 kB` | `131.201 kB`<br />`40.047 kB` |    `353 B` <br />`101 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `133.832 kB`<br />`40.479 kB` | `134.185 kB`<br />`40.577 kB` |     `353 B` <br />`98 B`  |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 | `117.083 kB`<br />`36.122 kB` | `118.471 kB`<br />`36.448 kB` | `1.388 kB` <br />`326 B`  |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                                                                    |   `24.249 kB`<br />`8.856 kB` |   `24.602 kB`<br />`8.964 kB` |    `353 B` <br />`108 B`  |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `47.119 kB`<br />`16.528 kB` |  `47.472 kB`<br />`16.632 kB` |    `353 B` <br />`104 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                |          Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                          |  `93.549 kB`<br />`22.537 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                       |  `57.797 kB`<br />`15.091 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                             |   `15.646 kB`<br />`6.298 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                     |  `73.973 kB`<br />`19.582 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> |   `64.899 kB`<br />`17.91 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>               |  `36.132 kB`<br />`11.954 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                                                    |  `64.718 kB`<br />`17.012 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                            |    `6.473 kB`<br />`2.196 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                    | `150.868 kB`<br />`41.518 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                               |  `45.111 kB`<br />`12.567 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                    | `133.356 kB`<br />`34.002 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                            |  `79.125 kB`<br />`19.379 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                      |  `78.455 kB`<br />`19.187 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/tag.fixture.js'>Tag</abbr>                                                                               |    `22.004 kB`<br />`7.93 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1d813ca54572065b0c19e3fbad9e81119c8ebc50'>1d813ca54572065b0c19e3fbad9e81119c8ebc50</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-positioning/src/createSlideStyles.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import { tokens } from '@fluentui/react-theme';
```
Comment (ling1726) [inline packages/react-components/react-positioning/src/createSlideStyles.ts:1]: Lives in the same package as the arrow styles
https://github.com/microsoft/fluentui/blob/8b699accbe90425a8f3042b0943ac20d457d9bcc/packages/react-components/react-positioning/src/createArrowStyles.ts#L63

File: packages/react-components/react-popover/src/components/PopoverSurface/usePopoverSurfaceStyles.styles.ts [hunk 2]
```diff
@@ -28,6 +28,8 @@ const useStyles = makeStyles({
     ...typographyStyles.body1,
   },
 
+  slideAndFadeIn: createSlideStyles(10),
```
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/react-components/react-popover/src/components/PopoverSurface/usePopoverSurfaceStyles.styles.ts:31]: better style encapsulation, are there any downsides to keeping it like this?

File: packages/react-components/react-positioning/src/createSlideStyles.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        fadeIn,
+        {
+          from: {
+            transform: `translate(${mainAxis}px, 0px)`,
+          },
+        },
+      ],
+    },
+    // FIXME: remove casting https://github.com/microsoft/griffel/issues/378
+  } as GriffelStyle;
```
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/react-components/react-positioning/src/createSlideStyles.ts]: Yup css vars work perfectly here, will remember that interesting usage pattern for the future 👍

**Review by layershifter — APPROVED** [review]

File: apps/vr-tests-react-components/src/stories/Menu/Menu.stories.tsx [hunk 4]
```diff
@@ -15,8 +15,10 @@ import {
   MenuSplitGroup,
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/Menu/Menu.stories.tsx:15]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 52 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=28119&commitId=196f45f2a77f97ee4bf94e202cb7ab2ea0894f3d&env=prod)

✅ There was 2 screenshots added, 27 screenshots removed, 1894 screenshots unchanged, 0 screenshots with different dimensions and 23 screenshots with visible difference.


  <details><summary><b>unknown</b> 52 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Menu Converged - basic.default - RTL.chromium.png|1861| Changed|
|Menu Converged - basic.default.chromium.png|1594| Changed|
|Menu Converged - groups.default - Dark Mode.chromium.png|40538| Changed|
|Menu Converged - groups.default - High Contrast.chromium.png|49307| Changed|
|Menu Converged - groups.default - RTL.chromium.png|3465| Changed|
|Menu Converged - groups.default.chromium.png|3163| Changed|
|Menu Converged - icon slotted content.default - RTL.chromium.png|2271| Changed|
|Menu Converged - icon slotted content.default.chromium.png|2011| Changed|
|Menu Converged - nested submenus.default - RTL.chromium.png|2476| Changed|
|Menu Converged - nested submenus.default.chromium.png|2207| Changed|
|Menu Converged - secondary content.default - RTL.chromium.png|2137| Changed|
|Menu Converged - secondary content.default.chromium.png|1869| Changed|
|Menu Converged - selection groups.default - RTL.chromium.png|3557| Changed|
|Menu Converged - selection groups.default.chromium.png|3251| Changed|
|Menu Converged - selection.checkbox - Dark Mode.chromium.png|36523| Changed|
|Menu Converged - selection.checkbox - High Contrast.chromium.png|43674| Changed|
|Menu Converged - selection.checkbox - RTL.chromium.png|1861| Changed|
|Menu Converged - selection.checkbox.chromium.png|1594| Changed|
|Menu Converged - split item.default - RTL.chromium.png|2308| Changed|
|Menu Converged - split item.default.chromium.png|2038| Changed|
|Menu Converged - submenuIndicator slotted content.default - RTL.chromium.png|1384| Changed|
|Menu Converged - submenuIndicator slotted content.default.chromium.png|1129| Changed|
|Menu nested within a MenuTrigger.default.chromium.png|3446| Changed|
|Menu.Nested Submenus Small Viewport Flipped.chromium.png|0| Added|
|Menu.Nested Submenus Small Viewport Stacked.chromium.png|0| Added|
|Menu Converged - basic.default - RTL.hover menuitem.chromium.png|0| Removed|
|Menu Converged - basic.default.hover menuitem.chromium.png|0| Removed|
|Menu Converged - icon slotted content.default - RTL.hover menuitem.chromium.png|0| Removed|
|Menu Converged - icon slotted content.default.hover menuitem.chromium.png|0| Removed|
|Menu Converged - nested submenus.default - RTL.all open.chromium.png|0| Removed|
|Menu Converged - nested submenus.default.all open.chromium.png|0| Removed|
|Menu Converged - secondary content.default - RTL.hover menuitem.chromium.png|0| Removed|
|Menu Converged - secondary content.default.hover menuitem.chromium.png|0| Removed|
|Menu Converged - selection groups.default - RTL.selected.chromium.png|0| Removed|
|Menu Converged - selection groups.default.selected.chromium.png|0| Removed|
|Menu Converged - selection.checkbox - Dark Mode.selected.chromium.png|0| Removed|
|Menu Converged - selection.checkbox - High Contrast.selected.chromium.png|0| Removed|
|Menu Converged - selection.checkbox - RTL.selected.chromium.png|0| Removed|
|Menu Converged - selection.checkbox.selected.chromium.png|0| Removed|
|Menu Converged - split item.default - RTL.submenu open.chromium.png|0| Removed|
|Menu Converged - split item.default.submenu open.chromium.png|0| Removed|
|Menu Converged - submenuIndicator slotted content.default - RTL.submenus open.chromium.png|0| Removed|
|Menu Converged - submenuIndicator slotted content.default.submenus open.chromium.png|0| Removed|
|Menu nested within a MenuTrigger.default.submenu open.chromium.png|0| Removed|
|Menu.Nested Submenus Small Viewport Flipped.default.chromium.png|0| Removed|
|Menu.Nested Submenus Small Viewport Flipped.nested menu.chromium.png|0| Removed|
|Menu.Nested Submenus Small Viewport Stacked.default.chromium.png|0| Removed|
|Menu.Nested Submenus Small Viewport Stacked.nested menu.chromium.png|0| Removed|
|Popover Converged.avoid scrolling.PopoverSurface focused.chromium.png|0| Removed|
|Popover Converged.positioning - High Contrast.positioned popovers.chromium.png|0| Removed|
|Popover Converged.positioning - RTL.positioned popovers.chromium.png|0| Removed|
|Popover Converged.positioning.positioned popovers.chromium.png|0| Removed|

</details>


<div id="vrtCommentfluentuiv9"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Comment by daisygeng** [comment]
thank you!!

**Pull request merged**